src/_pytest/_code/source.py

```diff
@@ -292,6 +293,7 @@ def getfslineno(obj) -> Tuple[Optional[Union["Literal['']", py.path.local]], int
 
     try:
         code = Code(obj)
+        return code.path, code.firstlineno
```

Comment:
I think it's better to keep this in the `else` branch of the `try`, to keep the `try` minimal (e.g. if `code.firstlineno` raises a `TypeError` for some reason, it should propagate, not be caught).